src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,364 @@
+        """Return a xx-Measurement or Spectrum with the data and metadata recorded from
+        a setup at SurfCat at given timestamp
+
+        All attributes of this reader can be accessed from the
+        measurement as `measurement.reader.attribute_name`.
+
+        Args:
+            path_to_file (str): Named argument from Measurement Class.
+                                Can be used as the setup name in the cinfdatabase
+            **kwargs (dict): Key-word arguments are passed to cinf Measurement.__init__
+                             setup_name (str): The setup name in the database default
+                                 to path_to_file
+                             timestamp (str): Timestamp the measurement started
+                                              given as (YYYY-MM-DD HH:MM:SS)
+        """
+
+        self.measurement_class = kwargs.pop("measurement_class", cls)
+        self.setup_name = kwargs.pop("setup_name", path_to_file)
+        self.timestamp = kwargs.pop("timestamp", None)
+        self.comment = kwargs.pop("comment", None)
+        self.mass_scans = kwargs.pop("include_mass_scans", False)
+        self.verbose = kwargs.pop("verbose", False)
+        self.grouping_column = kwargs.pop("group", None)
+
+        # figure out whether to collect data group by 'comment' or 'timestamp'
+        if not self.grouping_column:
+            if self.timestamp and not self.comment:
+                self.grouping_column = "time"
+                self.token = self.timestamp
+            elif self.comment and not self.timestamp:
+                self.grouping_column = "comment"
+                self.token = self.comment
+            else:
+                warnings.warn(
+                    "Both a comment and a timestamp is given "
+                    "but no explicit grouping column is set. \n"
+                    f"Defaults to 'time' using '{self.timestamp}'",
+                    stacklevel=2,
+                )
+                self.grouping_column = "time"
+                self.token = self.timestamp
+
+        plugins.activate_cinfdata()
+
+        if (
+            self.measurement_class == Spectrum
+            or self.measurement_class == SpectrumSeries
+            or self.measurement_class == MultiSpectrum
+        ):
+            return self.read_spectrums()
+
+        elif (
+            self.measurement_class == MSMeasurement
+            or self.measurement_class == ReactorMeasurement
+            or self.measurement_class == Measurement
+        ):
+            obj_as_dict = self.read_ms()
+            if issubclass(cls, self.measurement_class):
+                self.measurement_class = cls
+            obj_as_dict.update(kwargs)
+            self.measurement = self.measurement_class.from_dict(obj_as_dict)
+
+            if self.mass_scans:
+                if self.verbose:
+                    print("adding mass scans to the measurement")
+                self.measurement = self.add_mass_scans()
```

Comment:
Are you suggesting to put the method add_mass_scans() into the reactor technique / SpectroMSMeasurement / SpectroMeasurement?

or should I rename the method to download_associated_mass_scans?

Reply:
I have used that `self.measurement` can be accessed from the `add_mass_scans()` method and the SpectrumSeries of the mass scans downloaded are then added to measurement with '__add__'

`new_measurement = downloaded_mass_scans.__add__(self.measurement)` 

so thats why i called the method `add_mas_scans and()` not `add_and_overwrite`.

Reply:
What doesn't make sense to us is that it returns measurement, which is gramatically inconsistent with the name. 

For the name of the method to make sense, you could go with one of the following:
A. If add_mass_scans might be called seperately, and the measurement can be modified without making a new one, here it should read:
    `self.add_mass_scans(self.measurement)`
    where the argument to add_mass_scans is modified within the method
    OR
B. If add_mass_scans will only ever be called during this reader's .read(), the line here should simply be:
    `self.add_mass_scans()`
    where add_mass_scans contains a line like
    `self.measurement = new_measurement`

If you want to keep the method's returning a measurement, it should also take a measurement as an argument, and it should have a different name. This line would then accordingly become:
`self.measurement = self.with_mass_scans_added(self.measurement)`

Reply:
ahh. ok. I went with option B because I could not make option A behave properly. 
My limited coding skills cannot make that work.

All the options I have tried lead to the reader having a measurement that is updated and not the provided measurement. 

```
meas = Measurement.read(                                                     
        path_to_file=None, 
        timestamp=TIME,
        setup_name=SETUP,                           
        include_mass_scans = False,
        aliases={"pressure": ["Reactor pressure"], "temperature": ["TC temperature"]},
        technique='reactor',    
        reader='cinfdata_db'                                                 
        )  
```
Then `meas` is a measurement without mass scans.
To add mass scans one should then could do:

`meas.reader.add_mass_scans(meas)`

However, this lead to 

`meas.reader.measurement` 

as the updated measurement and not the original meas.